backend/src/relationships/friendships/friendships.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { EntityDoesNotExistError } from 'src/errors/entityDoesNotExist';
+import UserRepository from 'src/users/repository/user.repository';
+import { Repository } from 'typeorm';
+import { CreateFriendshipDto } from './dto/create-friendship.dto';
+import { QueryFriendshipDto } from './dto/query-friendship.dto';
+import { UpdateFriendshipDto } from './dto/update-friendship.dto';
+import { Friendship } from './entities/friendship.entity';
+import { User } from 'src/users/entities/user.entity';
+
+@Injectable()
+export class FriendshipsService {
+  constructor(
+    @InjectRepository(Friendship)
+    private friendshipRepository: Repository<Friendship>,
+    @InjectRepository(UserRepository)
+    private userRepository: UserRepository,
+  ) {}
+
+  async create(createFriendshipDto: CreateFriendshipDto) {
+    const friendship: Friendship = new Friendship();
+    friendship.source = await this.userRepository.findOne(
+      createFriendshipDto.sourceId,
+    );
+    friendship.sourceId = createFriendshipDto.sourceId;
+    if (
+      friendship.source === undefined ||
+      friendship.source.id !== friendship.sourceId
+    ) {
+      throw new EntityDoesNotExistError(
+        `User #${createFriendshipDto.sourceId}`,
+      );
+    }
+
+    friendship.target = await this.userRepository.findOne(
+      createFriendshipDto.targetId,
+    );
+    friendship.targetId = createFriendshipDto.targetId;
+    if (
+      friendship.target === undefined ||
+      friendship.target.id !== friendship.targetId
+    ) {
+      throw new EntityDoesNotExistError(
+        `User #${createFriendshipDto.targetId}`,
+      );
+    }
+
+    const ret = await this.friendshipRepository.save(friendship);
+    friendship.source.friendshipsIds.push(friendship.id);
+    this.userRepository.update(friendship.sourceId, friendship.source);
+    friendship.target.friendshipsIds.push(friendship.id);
+    this.userRepository.update(friendship.targetId, friendship.target);
+    await this.userRepository.save(friendship.source);
+    await this.userRepository.save(friendship.target);
+    return ret;
+  }
+
+  findAll(query?: QueryFriendshipDto) {
+    return this.friendshipRepository.find({ where: query });
+  }
+
+  findOne(id: number) {
+    return this.friendshipRepository.findOne(id);
+  }
+
+  update(id: number, updateFriendshipDto: UpdateFriendshipDto) {
+    return this.friendshipRepository.update(id, updateFriendshipDto);
+  }
```

Comment:
I can't get `PATCH` requests to work on `Friendships`. Each time I get:

```
{
  "generatedMaps": [],
  "raw": [],
  "affected": 0
}
```

Can you confirm on your end? Because I might be doing something wrong

Reply:
PATCH request on /friendships/1 with body:
```
{
  "id": 1,
  "status": 2
}
```
gets me the response:
```
{
  "generatedMaps": [],
  "raw": [],
  "affected": 1
}
```
And the update is indeed taken into account in the db. I will change the enum to strings as you mentioned elsewhere though.

Reply:
Yes, it's working now. I must have been on the wrong branch or making a bad request!